Gatekeeper addon e2e tests

JitenPalaparthi opened the pull request:
## What this PR does / why we need it
<!--
Add detailed explanation of what this PR does and why it is
needed.
-->
Contains  e2e tests for Gatekeeper addon along with few helper methods. All below mentioned steps are automated.
 - Step-1 ```tanzu package install gatekeeper.tce.vmware.com ``` 
 - Step-2  ```kubectl apply -f  ${TCE-REPO-DIR}addons/packages/gatekeeper/test/e2e/fixtures/constraint-template.yaml/constraint-template.yaml```
  - Step-3 Check required CRDs after applying the Step-2 file. ```  kubectl get crds | grep -i k8srequiredlabels ```
  - Step-4   ``` kubectl apply -f ${TCE-REPO-DIR}addons/packages/gatekeeper/test/e2e/fixtures/constraint.yaml```
  - Step-5  ``` kubectl create ns test ```  User cannot create a namespace without owner (The actual use case)
  - Step-6  `` kubectl apply -f ${TCE-REPO-DIR}addons/packages/gatekeeper/test/e2e/fixtures/test-namespace.yaml``` 
               - After applying the Step-6 file , name space should be created as the namespace will be created with owner name.
          
## Which issue(s) this PR fixes
<!--
Usage: `Fixes #<issue number>`, or `Fixes (paste link of issue)`.
-->
Fixes: # [1061](https://github.com/vmware-tanzu/tce/issues/1061)

## Describe testing done for PR
<!--
Example: Created vSphere workload cluster to verify change. 
-->
Tests done on Standalone cluster using ginkgo and go test
## Special notes for your reviewer
<!--
Add any things that reviewers should be aware of as they review
your PR.
Example: Please verify how I handled foo aligns with overall plan.
-->
N/A
## Does this PR introduce a user-facing change?
<!--
If no, just write "NONE" in the release-note block below.
If yes, a release note is required:
Enter your extended release note in the block below.
-->
N/A
```release-note

```

jiten-p committed:
New helper package that can contain required commands

JitenPalaparthi committed:
add gatekeeper e2e test suite

JitenPalaparthi committed:
add required yaml files in fixtures

JitenPalaparthi committed:
temp hack for package intall stdout issue

JitenPalaparthi committed:
add gatekeeper suite tests

JitenPalaparthi committed:
add gatekeeper tests

JitenPalaparthi committed:
add new method to format and get value

JitenPalaparthi committed:
fix array copy bug in cmdhelper

JitenPalaparthi committed:
add separate file for local utilities

JitenPalaparthi committed:
fix issues during namespace creation

JitenPalaparthi committed:
Merge branch 'vmware-tanzu:main' into gatekeeper-addon-e2e-tests

JitenPalaparthi committed:
fix additional gomega import

JitenPalaparthi committed:
fix spellings,names,comments

jpmcb reviewed (changes requested):
Awesome! Thanks for adding these

We are about to split up the top level go module into individual go modules in https://github.com/vmware-tanzu/tce/pull/735. This way, package and plugin authors can have greater ownership over their own modules and there won't be any top level dependencies. See this issue for dicsussions and details: https://github.com/vmware-tanzu/tce/pull/753

Can you update this PR with it's own go mod? And then we will be able to merge it after https://github.com/vmware-tanzu/tce/pull/735

JitenPalaparthi committed:
add license to each file

JitenPalaparthi committed:
add method to run and  kill long running commands through a channel

JitenPalaparthi committed:
refactor code

JitenPalaparthi committed:
fix gatekeeper version issue on package delete

JitenPalaparthi committed:
update CliRunnerChan method with new implementation

JitenPalaparthi committed:
add go.mod to cmdhelper package

JitenPalaparthi committed:
add go.mod to gatekeeper e2e tests

JitenPalaparthi committed:
add README.md file for Gatekeeper e2e tests

jpmcb reviewed (changes requested):
Awesome! Thanks for adding those

The markdown linter is failing. Can you run `make check` to ensure there is proper formatting of the readme file?

JitenPalaparthi committed:
fix README.md lint issues

JitenPalaparthi committed:
fix location of go.mod

- go.mod has been changed from cmdhelper to pkg/

JitenPalaparthi committed:
fix after go.mod file change

 update go.mod file after cmdhelper package go.mod file change

JitenPalaparthi committed:
Merge branch 'vmware-tanzu:main' into gatekeeper-addon-e2e-tests

Commit:
moved tests from e2e to test directory

JitenPalaparthi committed:
Merge branch 'vmware-tanzu:main' into gatekeeper-addon-e2e-tests

JitenPalaparthi committed:
Merge branch 'vmware-tanzu:main' into gatekeeper-addon-e2e-tests

Commit:
Merge branch 'gatekeeper-addon-e2e-tests' of https://github.com/JitenPalaparthi/tce into gatekeeper-addon-e2e-tests

Commit:
add gatekeeper addon package e2e tests

these tests are developed based on ginkgo framework

before the tests are run,cluster should be up and running

tanzu and kubectl cli must be available

Signed-off-by: Jiten Palaparthi <jpalaparthi@vmware.com>

joshrosso commented:
:wave: -- hello, we've decided to close stale PRs, but do acknowledge that this is something we may want to get in eventually. We noticed it currently exists in your fork. If you'd like to re-open a PR with updated E2E tests, we'd love to take a look at them!
